src/Promitor.Core.Scraping/ResouceTypes/StorageQueueScraper.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.Monitor.Fluent.Models;
+using Microsoft.Extensions.Logging;
+using Promitor.Core.Scraping.Configuration.Model;
+using Promitor.Core.Scraping.Configuration.Model.Metrics.ResourceTypes;
+using Promitor.Core.Telemetry.Interfaces;
+using Promitor.Integrations.AzureMonitor;
+using Promitor.Integrations.AzureStorageQueue;
+
+namespace Promitor.Core.Scraping.ResouceTypes
+{
+    public class StorageQueueScraper: Scraper<StorageQueueMetricDefinition>
+    {
+        private readonly AzureStorageQueueClient _azureStorageQueueClient;
+        public StorageQueueScraper(AzureMetadata azureMetadata, MetricDefaults metricDefaults, AzureMonitorClient azureMonitorClient, ILogger logger, IExceptionTracker exceptionTracker)
+            : base(azureMetadata, metricDefaults, azureMonitorClient, logger, exceptionTracker)
+        {
+            _azureStorageQueueClient = new AzureStorageQueueClient(logger);
+        }
+
+        protected override async Task<double> ScrapeResourceAsync(StorageQueueMetricDefinition metricDefinition, AggregationType aggregationType, TimeSpan aggregationInterval)
+        {
+            if (metricDefinition.AzureMetricConfiguration.MetricName == "Duration")
```

Comment:
Would be good to use a `switch` and add a case for `Duration` and one for `MessageCount` instead of assuming this.

Next to that we should have a default that throws a not supported exception to point out that we are not handling those yet.

```suggestion
switch (metricDefinition.AzureMetricConfiguration.MetricName)
	{
		case "Duration":
			return await _azureStorageQueueClient.GetLastMessageDurationAsync(metricDefinition.AccountName, metricDefinition.QueueName, metricDefinition.SasToken);
		case "MessageCount":
			return await _azureStorageQueueClient.GetQueueMessageCountAsync(metricDefinition.AccountName, metricDefinition.QueueName, metricDefinition.SasToken);
		default:
			throw new NotSupportedException($"Metric name '{metricDefinition.AzureMetricConfiguration.MetricName}' is not supported")
	}
```

Reply:
Since I'm removing duration it leaves us with only one metric. And the validation part makes sure that the configuration mentions the correct type of metric. We'll introduce switch, when we introduce duration.

Reply:
That makes sense indeed, sorry for the confusion!